### ISSUE TITLE
Feature request: beforeRegisterValidation Event

### ISSUE THREAD

**Issue by bombozama** [rainlab/user-plugin#422]
In many cases I've had to make fields such as name and surname mandatory upon registration. Adding an event to manipulate the post data, validation rules, messages and custom attribute names would be great. I'll issue a PR that does this.

**Comment by mjauvin** [comment]
Can't `model.beforeCreate / model.beforeValidate` event be used for this?

ref. https://octobercms.com/docs/api/model/beforecreate

**Comment by bombozama** [comment]
From what I have in the current code, it just does some custom validation instead of using the model's. That's why I would suggest adding an event. Your solution would be better though... using the RainLab\User\Models\User validation rules instead of this.

This extract is from the Account.php component in the onRegister method:
```php
            /*
             * Validate input
             */
            $data = post();

            if (!array_key_exists('password_confirmation', $data)) {
                $data['password_confirmation'] = post('password');
            }

            $rules = [
                'email'    => 'required|email|between:6,255',
                'password' => 'required|between:4,255|confirmed'
            ];

            if ($this->loginAttribute() == UserSettings::LOGIN_USERNAME) {
                $rules['username'] = 'required|between:2,255';
            }

            $validation = Validator::make($data, $rules);
            if ($validation->fails()) {
                throw new ValidationException($validation);
            }
```

**Comment by bombozama** [comment]
Nevermind. I see @LukeTowers fixed it in 7307e999d1540103776bf2f7e0e98f3641023df1 
Now it's fully extendable.
